oraclepaas/resource_mysql_service_instance.go

```diff
@@ -0,0 +1,662 @@
+						},
+						"mysql_collation": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+						},
+						"mysql_port": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.IntBetween(3200, 3399),
+							Default:      3306,
+						},
+						"mysql_username": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+						"mysql_password": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+						"shape": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"subnet": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						/* Couldn't get these to work with the current API. I've commented them out for now
+						"mysql_options" : {
+							Type: schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						}
+						"mysql_timezone" : {
+							Type: schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						}
+						*/
+						"snapshot_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"source_service_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"enterprise_monitor": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							ForceNew: true,
+						},
+						"enterprise_monitor_configuration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"em_agent_password": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										ForceNew:  true,
+										Sensitive: true,
+									},
+									"em_agent_username": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ForceNew:     true,
+										Sensitive:    true,
+										ValidateFunc: validation.StringLenBetween(2, 32),
+									},
+									"em_password": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										ForceNew:  true,
+										Sensitive: true,
+									},
+									"em_username": {
+										Type:      schema.TypeString,
+										Optional:  true,
+										ForceNew:  true,
+										Sensitive: true,
+									},
+									"em_port": {
+										Type:         schema.TypeInt,
+										Optional:     true,
+										ForceNew:     true,
+										Computed:     true,
+										ValidateFunc: validation.IntBetween(1, 65535),
+									},
+								},
+							},
+						},
+						// this comes from the service.
+						"connect_string": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
```

Comment:
We can remove the `Optional` tag since this is only being set in the read function and not configurable by the user

Reply:
Done.